src/main/java/io/lettuce/core/protocol/CommandHandler.java

```diff
@@ -422,6 +421,11 @@ private void writeSingleCommand(ChannelHandlerContext ctx, RedisCommand<?, ?, ?>
             TraceContext context = provider.getTraceContext();
 
             Tracer.Span span = tracer.nextSpan(context);
+
```

Comment:
It should be possible to move this assignment into `channelActive`. Since Lettuce 6, we no longer reuse `CommandHandler` across multiple channels.

Also, we should fix batch writes so that batched commands participate in tracing as well.

Reply:
@mp911de Sorry for the delay - I've moved it to `channelActive`. I still need a null check when invoking `endpoint` because for the initial connection span, the remote address isn't available yet.

Supporting batched commands would be great but is tricky for modeling - I've filed this issue in OpenTelemetry to discuss what could be a good approach

https://github.com/open-telemetry/opentelemetry-specification/issues/1548